sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -130,6 +130,10 @@ export type MessageSessionOptions = SessionManagerOptions &
  * Describes the receiver for a Message Session.
  */
 export class MessageSession extends LinkEntity {
+  /**
+   * @property {string} [receiverName] Internal AMQP receiver link name.
+   */
+  receiverName?: string;
```

Comment:
The fact that auto lock renewal has started, it means that a session receiver got created at some point in time.

Rhea receivers get created in the `init()` function.
`init()` function takes in `ReceiverOptions` which contains the name that gets used to create the rhea receiver.

For sessions, `init()` gets called only once for each `messageSession` instance. `this.name` is what is used in the `name` property in the `options` sent to create the rhea receiver.

Therefore, we can skip the need for a new variable `receiverNAme`, and directly use `this.name` when calling `this._context.managementClient!.renewSessionLock`

The main reason I am pushing for this is I rather reduce the number of variables that is tracking the name and have one source of truth instead

Reply:
Thanks for fixing this with https://github.com/Azure/azure-sdk-for-js/pull/2353/commits/cf69d0452fd044e7cf3537044ab1a23aedc2ab91